docs/proposals/timeseries-partitioning-in-compactor.md

```diff
@@ -0,0 +1,267 @@
+---
+title: "Timeseries Partitioning in Compactor"
+linkTitle: "Timeseries Partitioning in Compactor"
+weight: 1
+slug: timeseries-partitioning-in-compactor
+---
+
+- Author: @roystchiang
+- Reviewers:
+- Date: August 2022
+- Status: Proposed
+
+
+## Timeseries Partitioning in Compactor
+
+## Introduction
+
+The compactor is a crucial component in Cortex responsible for deduplication of replicated data, and merging blocks across multiple time intervals together. This proposal will not go into great depth with why the compactor is necessary, but aims to focus on how to scale the compactor as a tenant grows within a Cortex cluster.
+
+
+## Problem and Requirements
+
+Cortex introduced horizontally scaling compactor which allows multiple compactors to compact blocks for a single tenant, sharded by time interval. The compactor is capable of compacting multiple smaller blocks into a larger block, to reduce the the duplicated information in index. The following is an illustration of how the shuffle sharding compactor works, where each arrow represents a single compaction that can be carried out independently.
+![Current Implementation](/images/proposals/parallel-compaction-grouping.png)
+
+However, if the tenant is sending unique timeseries, the compaction process does not help with reducing the index size. Furthermore, this scaling of parallelism by time interval is not sufficient for a tenant with hundreds of millions of timeseries, as more timeseries means longer compaction time.
+
+Currently, the compactor is bounded by the 64GB index size, and having a compaction that takes days to complete simply is not sustainable. This time includes the time to download the blocks, merging of the timeseries, writing to disk, and finally uploading to object storage.
+
+The compactor is able to compact up to 400M timeseries within 12 hours, and will fail with the error of index exceeding 64GB. Depending on the number of labels and the size of labels, one might reach the 64GB limit sooner. We need a solution that is capable of:
+
+* handling the 64GB index limit
+* reducing the overall compaction time
+    * reducing the amount of data downloaded
+    * reducing the time required to compact
```

Comment:
The most important thing is to allow the compactor to continue scaling horizontally